src/frontend/apps/standalone_site/src/features/PortabilityRequests/components/PortabilityRequests.tsx

```diff
@@ -54,31 +113,78 @@ export const PortabilityRequests = ({
 
-  const sorts = [
-    {
-      id: '-created_on',
-      label: intl.formatMessage(
-        commonSortMessages.sortByDescendingCreationDate,
-      ),
-    },
+  const pagination = usePagination({
+    defaultPage: 1,
+    pageSize: ITEM_PER_PAGE,
+  });
+
+  const [sortModel, setSortModel] = useState<SortModel>([
     {
-      id: 'created_on',
-      label: intl.formatMessage(commonSortMessages.sortByAscendingCreationDate),
+      field: 'created_on',
+      sort: 'desc',
     },
-  ];
+  ]);
 
-  const [currentSort, setCurrentSort] = useState(sorts[0]);
-  const [currentPage, setCurrentPage] = useState(1);
-  const { isError, isLoading, data, refetch } = usePortabilityRequests({
-    offset: `${(currentPage - 1) * ITEM_PER_PAGE}`,
-    limit: `${ITEM_PER_PAGE}`,
-    ordering: currentSort.id,
-    state: state || '', // disable filter by default
-    for_playlist_id: for_playlist_id || '', // disable filter by default
-  });
+  let ordering = '';
+  if (sortModel.length) {
+    ordering = sortModel[0].field;
+    if (sortModel[0].sort === 'desc') {
+      ordering = `-${ordering}`;
+    }
+  }
+
+  const { isError, isLoading, data, refetch, isFetching } =
+    usePortabilityRequests({
+      offset: `${(pagination.page - 1) * ITEM_PER_PAGE}`,
+      limit: `${ITEM_PER_PAGE}`,
+      ordering,
+      state, // disable filter by default
+      for_playlist_id, // disable filter by default
+    });
 
   useLtiUserAssociationJwtQueryParam();
 
+  useEffect(() => {
+    pagination.setPagesCount(
+      data?.count ? Math.ceil(data?.count / pagination.pageSize) : 0,
+    );
+  }, [data?.count, pagination]);
+
+  const hasNoResult = !isError && data && !data?.count && !isLoading;
+  const hasResult = !isError && data && data?.count && !isLoading;
+
+  /**
+   * Clean the data to be displayed in the table
+   */
+  const rows = useMemo(
+    () =>
+      data?.results
+        ? data?.results.map((portabilityRequest) => ({
+            id: portabilityRequest.id,
+            created_on: `${new Date(
+              portabilityRequest.created_on,
+            ).toLocaleDateString(navigator.language)} ${new Date(
+              portabilityRequest.created_on,
+            ).toLocaleTimeString(navigator.language)}`,
+            portabilityRequest: intl.formatMessage(
+              messages.rowPortabilityRequestText,
+              {
+                from_playlist: portabilityRequest.from_playlist.title,
+                for_playlist: portabilityRequest.for_playlist.title,
+              },
+            ),
+            consumerSite: portabilityRequest.from_lti_consumer_site?.name,
+            fromUserEmail:
+              portabilityRequest?.from_user?.email ||
+              intl.formatMessage(messages.requestFromLtiUser),
+            updatedUserEmail: portabilityRequest?.updated_by_user?.email,
+            state: portabilityRequest.state,
+            can_accept_or_reject: portabilityRequest.can_accept_or_reject,
+          }))
+        : [],
```

Comment:
IMHO, this cleanup could deserve  its own function.

Reply:
What about that? https://github.com/openfun/marsha/pull/2311/commits/aa063d311d22e568a9256cd6ead3fbafe70800dc